File: golang/util/create_test_file.go
package main

import (
    "bufio"
    "fmt"
    "math/rand"
    "os"
    "strings"
    "strconv"
    "time"
)

func main() {
    // Number of lines to write
    numLines := 1000000000

    // Open file in write mode
    file, err := os.Create("output_file.txt")
    if err != nil {
        panic(err)
    }
    defer file.Close()

    // Use buffered writer to write lines in batches
    start := time.Now()
    writer := bufio.NewWriter(file)
    batchSize := 1000
    for i := 0; i < numLines; i += batchSize {
        lines := make([]string, batchSize)
        for j := 0; j < batchSize; j++ {
            lines[j] = strconv.FormatFloat(rand.Float64(), 'f', 6, 64) + "\n"
        }
        _, err := writer.WriteString(strings.Join(lines, ""))
        if err != nil {
            panic(err)
        }
    }
    // Flush buffer to disk
    err = writer.Flush()
    if err != nil {
        panic(err)
    }
    elapsed := time.Since(start)

    fmt.Printf("Total time taken: %v\n", elapsed)
}

File: python/create_test_file.py
import random
import time

# Number of lines to write
num_lines = 1000000000

# Size of the buffer in bytes
buffer_size = 1024 * 1024 * 100  # 100 MB

# Open file in write mode with a buffer
with open('output_file.txt', 'w', buffering=buffer_size) as f:
    # Write lines in batches
    start_time = time.time()
    batch_size = 1000000
    for i in range(0, num_lines, batch_size):
        # Generate batch of random lines
        lines = [str(random.random()) + '\n' for _ in range(batch_size)]
        # Write batch of lines to file
        f.writelines(lines)
        # Flush buffer to disk after each batch
        f.flush()
    end_time = time.time()

# Print total time taken to write the file
print(f"Total time taken: {end_time - start_time:.2f} seconds")

